### PR TITLE
SymPy 1.2 compatibility

### PR DESCRIPTION
**Pull request by richardotis**
Provides compatibility fixes for codegen changes, as well as a workaround for sympy/sympy#14933

### CODE REVIEW

**Comment by bocklund** [comment]
Looks like conda is trying to update to py37 for some reason, causing failures for py27 builds.

**Comment by bocklund** [comment]
@richardotis can you add some comments to the Piecewise patch that make it clearer what is patched and why?

**Comment by bocklund** [comment]
Is anything holding up merging this?

File: pycalphad/core/patched_piecewise.py [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+    if len(_args) == 1 and _args[0][-1] == True:
+        return _args[0][0]
+
+    newargs = []  # the unevaluated conditions
+    current_cond = set()  # the conditions up to a given e, c pair
+    # make conditions canonical
+    args = []
+    for e, c in _args:
+        if not c.is_Atom and not isinstance(c, Relational):
+            free = c.expr_free_symbols
```
Comment (richardotis) [inline pycalphad/core/patched_piecewise.py]: This is the only line that was changed compared to the sympy 1.2 base; originally it was `c.free_symbols`. Details in https://github.com/sympy/sympy/issues/14933

**Comment by richardotis** [comment]
There are no blockers to merging this. On a related note, we should probably think about running the test suite against sympy master every once in a while, to avoid getting blindsided by another release.

**Comment by bocklund** [comment]
I added a comment. If the tests pass on CI, I'll go ahead and merge and rebase the other PRs

**Pull request merged**